build-test.sh failing (MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded... Assembly with same name is already loaded)

I updated to tip last night.

Ubuntu 16.04 (previously working)

```
./build.sh
./build-test.sh
```
failed.
```
git clean -xdf
./build.sh
./build-test.sh
```
failed.
435 errors building managed assemblies.
```
./build-test.sh generatelayoutonly
```
succeeded (I didn't need the managed test assemblies....

Did your failures look like this, by any chance? If so I am seeing this too.
```
/home/andy/repos/coreclr/.packages/microsoft.net.compilers.toolset/3.2.0-beta2-19272-03/tasks/netcoreapp2.1/Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly /home/andy/repos/coreclr/.packages/microsoft.net.compilers.toolset/3.2.0-beta2-19272-03/build/../tasks/netcoreapp2.1/Microsoft.Build.Tasks.CodeAnalysis.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/home/andy/repos/coreclr/tests/src/CoreMangLib/system/delegate/VSD/OpenDelegate.csproj]
```

This is the error I am seeing
```
/home/stmaclea/git/coreclr/.packages/microsoft.net.compilers.toolset/3.2.0-beta2-19272-03/tasks/netcoreapp2.1/Microsoft.CSharp.Core.targets(59,5): error MSB4062: 
The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly 
/home/stmaclea/git/coreclr/.packages/microsoft.net.compilers.toolset/3.2.0-beta2-19272-03/build/../tasks/netcoreapp2.1/Microsoft.Build.Tasks.CodeAnalysis.dll. 
Assembly with same name is already loaded Confirm that the <UsingTask> declaration is 
correct, that the assembly and all its dependencies are available, and that the task contains 
a public class that implements Microsoft.Build.Framework.ITask. [/home/stmaclea/git/coreclr/tests/src/JIT/Methodical/Arrays/lcs/_dbglcsmixed.csproj]
```

It seems to match @AndyAyersMS error.

When trying to reproduce, it worked w/o error when running `./build-test.sh`, but reappeared after cleaning.

So perhaps (strangely) this works
```
git clean -xdf
./build.sh
./build-test.sh
# ignore errors
./build-test.sh generatelayoutonly
./build-test.sh
```

For reference b51993933 is what I downloaded last night.
I do not have a reference for the last working version. I updated from 96e8c58d77

Same here, I tried to run in on ARM64 and got following error:

```log
robox@ubunddarm64s-016:/home/adsitnik/coreclr$ ./build-test.sh release
__DistroRid: linux-arm64
__RuntimeId: linux-arm64
Using RID linux-arm64 for BuildTools native tools
Skipped installing build tools.
Building Tests...
__BuildOS: Linux
__BuildArch: arm64
__BuildType: Release
__TestIntermediatesDir: /home/adsitnik/coreclr/bin/tests/obj/Linux.arm64.Release
__NativeTestIntermediatesDir: /home/adsitnik/coreclr/bin/tests/obj/Linux.arm64.Release/Native
__ManagedTestIntermediatesDir: /home/adsitnik/coreclr/bin/tests/obj/Linux.arm64.Release/Managed
Creating TestBinDir: /home/adsitnik/coreclr/bin/tests/Linux.arm64.Release
Creating LogsDir: /home/adsitnik/coreclr/bin/Logs
Creating MsbuildDebugLogsDir: /home/adsitnik/coreclr/bin/Logs/MsbuildDebugLogs
Building step 'Restore product binaries (build tests)' via "/home/adsitnik/coreclr/dotnet.sh" msbuild /nologo /verbosity:minimal /clp:Summary /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true /p:UsePartialNGENOptimization=false /maxcpucount /home/adsitnik/coreclr/tests/build.proj "/flp:Verbosity=normal;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.log" "/flp1:WarningsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.wrn" "/flp2:ErrorsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.err" /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll\;LogFile=binclash.log /t:BatchRestorePackages /p:__BuildArch=arm64 /p:__BuildType=Release /p:__BuildOS=Linux
Running init-dotnet.sh
Installing dotnet using Arcade...
Running: /home/adsitnik/coreclr/.dotnet/dotnet msbuild /nologo /verbosity:minimal /clp:Summary /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true /p:UsePartialNGENOptimization=false /maxcpucount /home/adsitnik/coreclr/tests/build.proj /flp:Verbosity=normal;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.log /flp1:WarningsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.wrn /flp2:ErrorsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.err /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log /t:BatchRestorePackages /p:__BuildArch=arm64 /p:__BuildType=Release /p:__BuildOS=Linux
MSBUILD : error MSB1021: Cannot create an instance of the logger. The given assembly name or codebase was invalid. (0x80131047)
Switch: BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log
ERROR: An error occurred in /home/adsitnik/coreclr/.dotnet/dotnet msbuild /nologo /verbosity:minimal /clp:Summary /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true /p:UsePartialNGENOptimization=false /maxcpucount /home/adsitnik/coreclr/tests/build.proj /flp:Verbosity=normal;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.log /flp1:WarningsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.wrn /flp2:ErrorsOnly;LogFile=/home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.err /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log /t:BatchRestorePackages /p:__BuildArch=arm64 /p:__BuildType=Release /p:__BuildOS=Linux. Check logs under /home/adsitnik/coreclr.
Failed to build Restore product binaries (build tests). See the build logs:
    /home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.log
    /home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.wrn
    /home/adsitnik/coreclr/bin/Logs/Restore_Product.Linux.arm64.Release.err
```

The folder with logs is empty because the code fails when it tries to create the logger.

--- a/build-test.sh
+++ b/build-test.sh
@@ -930,7 +930,7 @@ else
   __NumProc=$(nproc --all)
 fi
 
-__CommonMSBuildArgs=("/p:__BuildArch=$__BuildArch" "/p:__BuildType=$__BuildType" "/p:__BuildOS=$__BuildOS")
+__CommonMSBuildArgs=("/p:__BuildArch=$__BuildArch" "/p:__BuildType=$__BuildType" "/p:__BuildOS=$__BuildOS" "/nodeReuse:false")
 
 # Configure environment if we are doing a verbose build
 if [ $__VerboseBuild == 1 ]; then

--- a/build.sh
+++ b/build.sh
@@ -970,7 +970,7 @@ while :; do
     shift
 done
 
-__CommonMSBuildArgs="/p:__BuildArch=$__BuildArch /p:__BuildType=$__BuildType /p:__BuildOS=$__BuildOS "/nodeReuse:false" $__OfficialBuildIdArg $__SignTypeArg $__SkipRestoreArg"
+__CommonMSBuildArgs="/p:__BuildArch=$__BuildArch /p:__BuildType=$__BuildType /p:__BuildOS=$__BuildOS /nodeReuse:false $__OfficialBuildIdArg $__SignTypeArg $__SkipRestoreArg"
 
 # Configure environment if we are doing a verbose build
 if [ $__VerboseBuild == 1 ]; then

--- a/build.sh
+++ b/build.sh
@@ -970,7 +970,7 @@ while :; do
     shift
 done
 
-__CommonMSBuildArgs="/p:__BuildArch=$__BuildArch /p:__BuildType=$__BuildType /p:__BuildOS=$__BuildOS $__OfficialBuildIdArg $__SignTypeArg $__SkipRestoreArg"
+__CommonMSBuildArgs="/p:__BuildArch=$__BuildArch /p:__BuildType=$__BuildType /p:__BuildOS=$__BuildOS "/nodeReuse:false" $__OfficialBuildIdArg $__SignTypeArg $__SkipRestoreArg"
 
 # Configure environment if we are doing a verbose build
 if [ $__VerboseBuild == 1 ]; then

